batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -124,6 +127,25 @@ pub async fn receive(
             Ok(data) => data,
             Err(e) => {
                 warn!("Error while handling batcher response: {:?}", e);
+                // When submitting multiple batches, an InsufficientBalance error may occur, when the required balance of a user would exceed his balance in the BatcherPaymentService.sol
+                // This leads to a scenario where some of the submitted proofs are accepted and others rejected, with
+                // the SubmitError::InsufficientBalance(error_nonce) thrown. To ensure the user is notified that some of their proofs were rejected,
+                // we return upon erroring the nonce of the proof that has errored and set the new `last_proof_nonce` value accordingly.
+                // This ensures the client messaging protocol continues receiving verification and error responses until all messages are received.
+                if let SubmitError::InsufficientBalance(_, error_nonce) = e {
+                    aligned_submitted_data.push(Err(e));
+
+                    let last_valid_nonce = error_nonce - 1;
```

Comment:
what happens when the first message is insufficient balance? maybe `last_valid_nonce` will underflow? (0-1 or similar)